modules/core/04-channel/types/msgs.go

```diff
@@ -517,18 +517,97 @@ var _ sdk.Msg = &MsgChannelUpgradeTry{}
 
 // NewMsgChannelUpgradeTry constructs a new MsgChannelUpgradeTry
 // nolint:interfacer
-func NewMsgChannelUpgradeTry() *MsgChannelUpgradeTry {
-	return &MsgChannelUpgradeTry{}
+func NewMsgChannelUpgradeTry(
+	portID, channelID string,
+	counterpartyChannel Channel,
+	counterpartySequence uint64,
+	proposedUpgradeChannel Channel,
+	timeoutHeight clienttypes.Height,
+	timeoutTimestamp uint64,
+	proofChannel []byte,
+	proofUpgradeTimeout []byte,
+	proofUpgradeSequence []byte,
+	proofHeight clienttypes.Height,
+	signer string,
+) *MsgChannelUpgradeTry {
+	return &MsgChannelUpgradeTry{
+		PortId:                 portID,
+		ChannelId:              channelID,
+		CounterpartyChannel:    counterpartyChannel,
+		CounterpartySequence:   counterpartySequence,
+		ProposedUpgradeChannel: proposedUpgradeChannel,
+		TimeoutHeight:          timeoutHeight,
+		TimeoutTimestamp:       timeoutTimestamp,
+		ProofChannel:           proofChannel,
+		ProofUpgradeTimeout:    proofUpgradeTimeout,
+		ProofUpgradeSequence:   proofUpgradeSequence,
+		ProofHeight:            proofHeight,
+		Signer:                 signer,
+	}
 }
 
 // ValidateBasic implements sdk.Msg
 func (msg MsgChannelUpgradeTry) ValidateBasic() error {
-	return errors.New("error method not implemented")
+	if err := host.PortIdentifierValidator(msg.PortId); err != nil {
+		return sdkerrors.Wrap(err, "invalid port ID")
+	}
+
+	if !IsValidChannelID(msg.ChannelId) {
+		return ErrInvalidChannelIdentifier
+	}
+
+	if msg.CounterpartyChannel.State != INITUPGRADE {
+		return sdkerrors.Wrapf(ErrInvalidChannelState, "expected: %s, got: %s", INITUPGRADE, msg.CounterpartyChannel.State)
+	}
+
+	if msg.CounterpartySequence == 0 {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidSequence, "counterparty sequence cannot be 0")
+	}
+
+	if msg.ProposedUpgradeChannel.State != TRYUPGRADE {
+		return sdkerrors.Wrapf(ErrInvalidChannelState, "expected: %s, got: %s", TRYUPGRADE, msg.CounterpartyChannel.State)
+	}
+
+	if msg.TimeoutHeight.IsZero() {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidHeight, "timeout height must be non-zero")
+	}
+
+	if msg.TimeoutTimestamp == 0 {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidHeight, "timeout timestamp must be non-zero")
+	}
```

Comment:
I think these checks should grouped with a logical AND:

```go
if msg.TimeoutHeight.IsZero() && msg.TimeoutTimestamp == 0 {
	return sdkerrors.Wrap(sdkerrors.ErrInvalidUpgradeTimeout, "either timeout height or timeout timestamp must be non-zero")
}
```

We could add a new channel error `ErrInvalidUpgradeTimeout`